src/utils/useLocaleStorage.ts

```diff
@@ -0,0 +1,42 @@
+import { useState } from 'react';
+import { MakeChanges } from '../types/MakeChanges';
+import { Todo } from '../types/Todo';
+
+export const useLocaleStorage = (key: string): [Todo[], MakeChanges] => {
+  const [value, setValue] = useState<Todo[]>(() => {
+    return JSON.parse(window.localStorage.getItem(key) || '[]');
+  });
+
+  const setData = (data: Todo[]) => {
+    setValue(data);
+    window.localStorage.setItem(key, JSON.stringify(data));
+  };
+
+  const makeChange = {
+    add(data: Todo) {
+      setData([...value, data]);
+    },
```

Comment:
if you need to take the previous State and modify it, you need to the callback for setData, please check this link: https://stackoverflow.com/questions/55823296/reactjs-prevstate-in-the-new-usestate-react-hook